File: Factory/FactoryMethod/Java/src/PizzaFriday.java
package eu.svetlin.design.patterns.factory.factorymethod;

import eu.svetlin.design.patterns.factory.factorymethod.Pizza.Pizza;
import eu.svetlin.design.patterns.factory.factorymethod.PizzaStore.ChicagoPizzaStore;
import eu.svetlin.design.patterns.factory.factorymethod.PizzaStore.NewYorkPizzaStore;
import eu.svetlin.design.patterns.factory.factorymethod.PizzaStore.PizzaStore;

public class PizzaFriday {
   public static void main(String[] args){
        PizzaStore newYorkStore = new NewYorkPizzaStore();
        PizzaStore chicagoStore = new ChicagoPizzaStore();

        Pizza pizza = newYorkStore.orderPizza("cheese");
        System.out.println("Ethan ordered a " + pizza.getName() + "\n");

        pizza = chicagoStore.orderPizza("pepperoni");
        System.out.println("Joel ordered a " + pizza.getName() + "\n");
    }
}

File: Factory/FactoryMethod/TypeScript/src/PizzaFriday.ts
import Pizza from './Pizza/Pizza';
import NewYorkPizzaStore from './PizzaStore/NewYorkPizzaStore';
import ChicagoPizzaStore from './PizzaStore/ChicagoPizzaStore';

const pizzaFriday = () => {
  const newYorkPizzaStore = new NewYorkPizzaStore();
  const chicagoPizzaStore = new ChicagoPizzaStore();

  let pizza: Pizza = newYorkPizzaStore.orderPizza('cheese');
  console.log(`Ethan ordered a ${pizza.getName()}\n`);

  pizza = chicagoPizzaStore.orderPizza('pepperoni');
  console.log(`Joel ordered a ${pizza.getName()}\n`);

  console.log(pizza.toString());
};

pizzaFriday();
